Source/SwiftLintFramework/Rules/Style/RacistDeclarationRule.swift

```diff
@@ -0,0 +1,72 @@
+import SourceKittenFramework
+
+public struct RacistDeclarationRule: ASTRule, ConfigurationProviderRule, OptInRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "racist_declaration",
+        name: "Racist Declaration",
+        description: "Terms that have racist connotations should not be used in declarations.",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("let foo = \"abc\""),
+            Example("""
+            enum AllowList {
+                case foo, bar
+            }
+            """),
+            Example("func updateAllowList(add: String) {}")
+        ],
+        triggeringExamples: [
+            Example("let ↓slave = \"abc\""),
+            Example("""
+            enum ↓BlackList {
+                case foo, bar
+            }
+            """),
+            Example("func ↓updateWhiteList(add: String) {}"),
+            Example("""
+            enum ListType {
+                case ↓whitelist
+                case ↓blacklist
+            }
+            """),
+            Example("↓init(master: String, slave: String) {}"),
+            Example("""
+            final class FooBar {
+                func register<↓Master, ↓Slave>(one: Master, two: Slave) {}
+            }
+            """)
+        ]
+    )
+
+    private let racistTerms: Set<String> = [
+        "whitelist",
+        "blacklist",
+        "master",
+        "slave"
```

Comment:
Are you sure these are the terms you want to discourage? Are you OK to use the words listed [here](https://en.wikipedia.org/wiki/List_of_ethnic_slurs) in your source code, including the N-word?

Reply:
The point is to stop using words that up until now have been commonly used in source code. To provide a gentle nudge or reminder to developers that accidentally use the old terminology to use the new terminology.

Reply:
Could you also try to justify your design with a hard-coded array in Swift code?

Reply:
This isn't my design but the reason a hardcoded array makes sense for this case is that every tech company in the US is currently going through their source code to eliminate exactly those 4 words:

https://developer-tech.com/news/2020/jun/15/github-replace-slavery-terms-master-whitelist/
https://tools.ietf.org/id/draft-knodel-terminology-00.html#rfc.section.1.1

While making the words per-project configurable in swiftlint.yml would technically allow every company to solve the problem, every company is removing the same exact language right now, so having those 4 words seems like a very sensible default to me. I could see a case for allowing users to allow additional words via swiftlint.yml but I don't see any reason to make every project independently specify the same 4 words.

Reply:
I think there is room for both here. The term `racist_declaration` is combative where something like `disallowed_terminology` would be more likely to gain support from the individuals who don't have an opinion either way on the usage of the words outlined here. I do agree that the terms on this list cause conflict, and do have a place here as a set of default terms.

However, expanding this rule to `disallowed_terminologoy` would also allow more flexibility for projects to adopt their own particular rulesets. For a particular library use case, I might want to avoid certain terms or phrases in the source code that reference a particular company/technology stack/misnomer/etc. 

Expanding the scope of this rule enables that use case, which might help with the adoption.

Reply:
Totally agree. `disallowed_terminology` reads much better. Then maybe the default list would be those 4 words as well as an option to expand upon that list in their own projects?